### ISSUE TITLE
Allow to specify FieldInfo for a shadow property in Fluent API

### ISSUE THREAD

**Issue by AndriySvyryd** [aspnet/EntityFrameworkCore#15307]

**Comment by ajcvickers** [comment]
We think this might mean that when a property is configured as a shadow property, it can later get a `FieldInfo` and become a field-backed non-shadow property.

**Comment by divega** [comment]
We made the decision that during EF Core 3.0 we would re-enable configuration like this, after breaking it in preview 4. From the [breaking change documentation](https://docs.microsoft.com/ef/core/what-is-new/ef-core-3.0/breaking-changes#field-only-property-names-should-match-the-field-name):

> #### Mitigations
> Field-only properties must be named the same as the field they are mapped to. In a later preview of EF Core 3.0, we plan to re-enable explicitly configuring a field name that is different from the property name:
> ``` c#
> modelBuilder
>    .Entity<Blog>()
>    .Property("Id")
>    .HasField("_id");
> ```

Unfortunately we ended up punting this past 3.0.

**Comment by hisuwh** [comment]
This issue is closed but this does not seem to be fixed.  Is this a valid workaround:
```csharp
modelBuilder
   .Entity<Blog>()
   .Property("_id")
   .HasColumnName("Id");
```

**Comment by AndriySvyryd** [comment]
@hisuwh The issue was fixed in 3.1.11, if you are still experiencing it in 5.0.0 or later please open a new issue.

The proposed workaround is correct.

**Comment by hisuwh** [comment]
I'm using `3.1.16` and am still experiencing this issue.  I'm not in a position to move to 5.0.0 yet.
